core/embed/rust/src/ui/model_tt/component/hs_render.rs

```diff
@@ -0,0 +1,578 @@
+            let g = (((totals[GREEN_IDX][x] as u32 * BLUR_DIV) >> 13) & 0x07E0) as u16;
+            let b = (((totals[BLUE_IDX][x] as u32 * BLUR_DIV) >> 19) & 0x001F) as u16;
+            r | g | b
+        };
+
+        for i in 0..HOMESCREEN_IMAGE_SCALE as usize {
+            let j = 2 * (HOMESCREEN_IMAGE_SCALE as usize * x + i) as usize;
+            img_buffer.buffer[j + 1] = (c0 >> 8) as u8;
+            img_buffer.buffer[j] = (c0 & 0xFF) as u8;
+        }
+    }
+
+    let done = homescreen_get_fg_text(y, text_info, text_buffer, t_buffer);
+    homescreen_get_fg_icon(y, text_info, icon_data, t_buffer);
+
+    dma2d_wait_for_transfer();
+    dma2d_setup_4bpp_over_16bpp(text_info.2.into());
+    dma2d_start_blend(&t_buffer.buffer, &img_buffer.buffer, WIDTH);
+
+    done
+}
+
+fn homescreen_line(
+    icon_data: &mut [u8],
+    text_buffer: &mut BufferText,
+    text_info: HomescreenTextInfo,
+    image_data: &mut [u8; (HOMESCREEN_IMAGE_SIZE * 2) as usize],
+    y: i16,
+) -> bool {
+    let t_buffer = unsafe { get_buffer_4bpp((y & 0x1) as u16, true) };
+    let mut img_buffer = unsafe { get_buffer_16bpp((y & 0x1) as u16, false) };
+
+    for x in 0..HOMESCREEN_IMAGE_SIZE {
+        let x0 = (2 * x) as usize;
+        let x1 = x0 + 1;
+        let hi = image_data[x1];
+        let lo = image_data[x0];
+
+        let c = if y > HOMESCREEN_DIM_START {
+            let coef = 65536
+                - (((y - HOMESCREEN_DIM_START) as u32)
+                    * ((65536_f32 * HOMESCREEN_DIM) as u32 / HOMESCREEN_DIM_HIEGHT as u32));
+
+            let r = hi & 0xF8;
+            let g = ((hi & 0x07) << 5) | ((lo & 0xE0) >> 3);
+            let b = (lo & 0x1F) << 3;
+
+            let r = (((coef * r as u32) >> 8) & 0xF800) as u16;
+            let g = (((coef * g as u32) >> 13) & 0x07E0) as u16;
+            let b = (((coef * b as u32) >> 19) & 0x001F) as u16;
+            r | g | b
+        } else {
+            (hi as u16) << 8 | lo as u16
+        };
+
+        for i in 0..HOMESCREEN_IMAGE_SCALE {
+            let j = 2 * (HOMESCREEN_IMAGE_SCALE * x + i) as usize;
+            img_buffer.buffer[j + 1] = (c >> 8) as u8;
+            img_buffer.buffer[j] = (c & 0xFF) as u8;
+        }
+    }
+
+    let done = homescreen_get_fg_text(y, text_info, text_buffer, t_buffer);
+    homescreen_get_fg_icon(y, text_info, icon_data, t_buffer);
+
+    dma2d_wait_for_transfer();
+    dma2d_setup_4bpp_over_16bpp(text_info.2.into());
+    dma2d_start_blend(&t_buffer.buffer, &img_buffer.buffer, WIDTH);
+
+    done
+}
+
+fn homescreen_next_text(
+    texts: &Vec<Option<HomescreenText>, 4>,
+    text_buffer: &mut BufferText,
+    icon_data: &mut [u8],
+    text_info: HomescreenTextInfo,
+    text_idx: usize,
+) -> (HomescreenTextInfo, usize) {
+    let mut next_text_idx = text_idx;
+    let mut next_text_info = text_info;
+
+    if next_text_idx < texts.len() {
+        if let Some(txt) = texts[next_text_idx] {
+            unsafe { get_text_buffer(0, true) };
```

Comment:
Does this do anything?

Reply:
yeah it clears the buffer. changed it to fill(0) https://github.com/trezor/trezor-firmware/commit/eeb1516dd3a650ad3297440c9ad0baeaec04bfda